api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java

```diff
@@ -31,6 +31,7 @@
 	private Integer minLength; // If > 0, will always return identifiers with a minimum of this length
 	private Integer maxLength; // If > 0, will always return identifiers no longer than this length
     private String baseCharacterSet; // Enables configuration in appropriate Base
+	private Integer nextSequenceValue; // Pre-generate the next identifier
 
```

Comment:
Okay---makes sense as a use case.

Is the value of adding nextSequenceValue as an actual property on the objet that you can then update it via a standard OpenMRS REST call?  I still think, though a little more work, it would be safer to make a custom REST call that hits the DB (via identifierSourceService.saveSequenceValue) to update the nextSequenceValue without having to expose nextSequenceValue on the object itself.

I'm not an expert of the intricacies of Hibernate, but   SequentialIdentifierGeneratorProcessor.getIdentifiers() will change the underlying value of nextSequenceValue in the DB without updating the actual nextSequenceValue property of the IdentifierSource object.  I suppose if the source is never dirtied, we are okay, but seems like there's potential for the direct-to-DB update to be overridden on a flush.

@djazayeri, any thoughts?